org.eclipse.jdt.core/dom/org/eclipse/jdt/internal/core/dom/rewrite/ASTRewriteAnalyzer.java

```diff
@@ -913,6 +919,42 @@ private int rewriteJavadoc(ASTNode node, StructuralPropertyDescriptor property)
 		return pos;
 	}
 
+	private void rewriteStringTemplateNode(StringTemplateExpression node, int pos) {
+		RewriteEvent event= getEvent(node, StringTemplateExpression.MULTI_LINE);
+
+		if (event != null) {
+			switch (event.getChangeKind()) {
+				case RewriteEvent.REPLACED: {
+					boolean originalNode= ((Boolean) event.getOriginalValue()).booleanValue();
+					boolean newValue= ((Boolean) event.getNewValue()).booleanValue();
+
+					int pos1 = rewriteRequiredNode(node, StringTemplateExpression.TEMPLATE_PROCESSOR);
+					int pos2 = rewriteRequiredNode(node, StringTemplateExpression.STRING_TEMPLATE_COMPONENTS);
+
+					if(newValue && !originalNode) {// SINGLE LINE to MULTI LINE
+						if(pos2 == 0 ) { // without component
+							doTextReplace(pos1 +1, 1, "\"\"\"\n", getEditGroup(event)); //$NON-NLS-1$
+							doTextReplace(pos -1, 1, "\n\"\"\"", getEditGroup(event)); //$NON-NLS-1$
+						} else { // with Component
+							doTextReplace(pos1 +1, 1, "\"\"\"\n", getEditGroup(event)); //$NON-NLS-1$
```

Comment:
this code can be simplified to be one with appropriate pos assignments

Reply:
Done